dali/operators/math/expressions/arithmetic_test.cu

```diff
@@ -0,0 +1,241 @@
+          int sample_size_ = 1024 * 1024>
+struct ArithmOpParams {
+  static constexpr ArithmeticOp op = op_;
+  using Result = Result_;
+  using Left = Left_;
+  using Right = Right_;
+  static constexpr int IsLeftTensor = IsLeftTensor_;
+  static constexpr int IsRightTensor = IsRightTensor_;
+  static constexpr int blocks_x = blocks_x_;
+  static constexpr int thread_num = thread_num_;
+  static constexpr int batch_size = batch_size_;
+  static constexpr int tile_size = tile_size_;
+  static constexpr int sample_size = sample_size_;
+  static constexpr int tiles_per_sample = sample_size / tile_size;
+  static constexpr int num_tiles = batch_size * tiles_per_sample;
+  static_assert(sample_size >= tile_size, "This test doesn't support samples smaller than tiles.");
+};
+
+template <typename TestConfig>
+struct BinaryArithmeticOpGpuPerfTest : public ::testing::Test {
+  void SetUp() override {
+    stream = CUDAStream::Create(true);
+
+    /// Fill tile descriptors (shapes)
+    tile_descs.resize(TestConfig::num_tiles);
+    for (int sample_id = 0; sample_id < TestConfig::batch_size; sample_id++) {
+      for (int extent_id = 0; extent_id < TestConfig::tiles_per_sample; extent_id++) {
+        int tile_id = sample_id * TestConfig::tiles_per_sample + extent_id;
+        tile_descs[tile_id].sample_idx = sample_id;
+        tile_descs[tile_id].extent_idx = extent_id;
+        tile_descs[tile_id].tile_size = TestConfig::tile_size;
+        tile_descs[tile_id].extent_size = TestConfig::tile_size;
+      }
+    }
+
+    // Reshape memory for those tiles
+    result.reshape(uniform_list_shape<1>(TestConfig::batch_size,
+                                         {TestConfig::tile_size * TestConfig::tiles_per_sample}));
+    if (TestConfig::IsLeftTensor) {
+      left.reshape(uniform_list_shape<1>(TestConfig::batch_size,
+                                         {TestConfig::tile_size * TestConfig::tiles_per_sample}));
+    } else {
+      left.reshape(uniform_list_shape<1>(TestConfig::batch_size, {1}));
+    }
+    if (TestConfig::IsRightTensor) {
+      right.reshape(uniform_list_shape<1>(TestConfig::batch_size,
+                                          {TestConfig::tile_size * TestConfig::tiles_per_sample}));
+    } else {
+      right.reshape(uniform_list_shape<1>(TestConfig::batch_size, {1}));
+    }
+
+    Left l{};
+    Right r{};
+    auto fill_left = [&l]() { return l += 1; };
+    auto fill_right = [&r]() { return r += 1; };
+    Fill(left.cpu(), fill_left);
+    Fill(right.cpu(), fill_right);
+
+    // Fill pointers for tiles
+    tiles_data.reshape(uniform_list_shape<1>(1, {TestConfig::num_tiles}));
+    auto tiles_cpu = tiles_data.cpu()[0];
+    for (int sample_id = 0; sample_id < TestConfig::batch_size; sample_id++) {
+      for (int extent_id = 0; extent_id < TestConfig::tiles_per_sample; extent_id++) {
+        int tile_id = sample_id * TestConfig::tiles_per_sample + extent_id;
+        tiles_cpu(tile_id)->desc = tile_descs[tile_id];
```

Comment:
Can't you fill `tile_descs` here as well?

Reply:
I could but implementing that lambda would be probably a bit more of a hassle than a direct loop.